src/azul/terra.py

```diff
@@ -278,6 +284,31 @@ def insufficient_access(self, resource: str):
         )
 
 
+class TerraClientException(Exception):
+    pass
+
+
+class TerraTimeoutException(TerraClientException):
+
+    def __init__(self, url: str, timeout: float):
+        super().__init__(f'No response from {url!r} within {timeout} seconds')
+
+
+class TerraStatusException(TerraClientException):
+
+    def __init__(self, url: str, response: HTTPResponse):
+        super().__init__(f'Unexpected response from {url!r}',
+                         response.status,
+                         response.data)
+
+
+class TerraNameConflictException(TerraClientException):
+
+    def __int__(self, url: str, source_name: str, response):
+        super().__init__(f'More than one source from {url!r} matched the name '
+                         f'{source_name!r}', response)
```

Comment:
W

```suggestion
        super().__init__(f'More than one source named {source_name!r}', 
        url, response)
```

but indented correctly.